### PR TITLE
feat: ColorPicker show clear when value is not set

### LINKED ISSUE #45517: ColorPicker 默认值无法为空
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-wf7q8k?file=/index.js)

### Steps to reproduce

1、没有给color 设置值，但是页面能到 有值 #1677FF, 点击按钮会打印form ，color 值为undefined 
2、color 为非必填，选择之后，无法再次去掉颜色， color 永远有值

### What is expected?

初始化时 RGB 可以为空，并且 选择之后可以清空

### What is actually happening?

RGB 为undefined,但是 ColorPicker 给了一个默认值，并且无法清空，清空回调函数也未执行

| Environment | Info |
| --- | --- |
| antd | 5.10.1 |
| React | 18 |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- close #45517
- #45456

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     ColorPicker show clear when value is not set      |
| 🇨🇳 Chinese |    颜色选择器在没有设置值时显示清空状态       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at d0c65c5</samp>

Refactored the `ColorPicker` component to remove unnecessary hook and improve clear icon logic. Updated and added test cases for the component. Modified all the demos for the component to use the `useToken` hook and the theme color as the default color.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at d0c65c5</samp>

*  Removed `useToken` hook from `ColorPicker` component and passed empty string as default color value to `useColorState` hook ([link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL110-R110))
*  Initialized `colorCleared` state with a boolean value that indicates whether the color picker has no value or defaultValue, to avoid showing the clear icon when the color picker is empty by default ([link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL127-R125))
*  Modified test cases for `allowClear`, `onClear`, `onChange`, and `showText` props to pass `defaultValue` prop to `ColorPicker` component, to make the clear icon and the text visible ([link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L117-R119), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L329-R333), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L375-R381), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L382-R388))
*  Added a new test case to check that the clear icon shows up when the `ColorPicker` component has no value or defaultValue, as this is the new behavior introduced by the previous change ([link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4R604-R608))
*  Modified demos for various props and features of the `ColorPicker` component to import the `theme` module and use the `useToken` hook to get the default color value from the theme, and pass it as the `defaultValue` or `value` prop to the `ColorPicker` component, to demonstrate how to use the theme color as the default or initial color for the color picker ([link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-f88c7b3daab6357fa92d93c9b3d2547083b32be4de4d4cb4c84cd3b6a546bac7L2-R9), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-c9f8c74dc86457938053a55bd0dc68780cb138694afe083e6a5510e26a76f32fL2-R9), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-2ca0011d2c022208815c98d2c5d0588715d3a6f8ba4c2ea858aea5c659f4d5c8L2-R10), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-ee31bf26c76769bc24652b314e1f09ab55e86fbba9ef227141569e16cff8e68bL2-R9), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-3a18ac72d96eecdc3a6adf0e3931167012ba8d1796a66da1c1b0e8a48fffd0ebL2-R9), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-e00c41c079465e93600695586a06b1a5777aad2bf2ca30ea7f97c7a953a58550L2-R130), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-223d7ae6b7c1e189b58f0a96c353e9a5016878eb74b7675fff666216f9bd828dL2-R54), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-9857979bb39f8140a694fe3b7f8e87e583bbd4291c0ffe3c14b461b59eaec2adL2-R22), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-ed290e827ebf9e0a1135a4c2ac312b932b674d9551f7468adcc5e242f4391405L1-R18), [link](https://github.com/ant-design/ant-design/pull/45993/files?diff=unified&w=0#diff-ed47d85df8868cab33de153e85588979000246ff0a11983fa0b343cb2ccd2690L2-R9))

### CODE REVIEW

File: components/color-picker/demo/disabled.tsx [hunk 1]
```diff
@@ -1,4 +1,9 @@
...
-import { ColorPicker } from 'antd';
+import { ColorPicker, theme } from 'antd';
 
-export default () => <ColorPicker showText disabled />;
+export default () => {
+  const {
+    token: { colorPrimary },
+  } = theme.useToken();
+  return <ColorPicker defaultValue={colorPrimary} showText disabled />;
+};
```
Comment (afc163) [inline components/color-picker/demo/disabled.tsx:9]: 为啥要这样改，感觉这样会让 demo 代码变得复杂，大多少业务是不是不需要 useToken ？

Comment (RedJue) [reply components/color-picker/demo/disabled.tsx:9]: 只是为了设置个初始颜色，现在改完之后没有默认颜色会显示清除状态，不用token的话就直接写死色值么？

File: components/color-picker/demo/allowClear.tsx [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import React from 'react';
-import { ColorPicker } from 'antd';
+import { ColorPicker, theme } from 'antd';
 
-export default () => <ColorPicker allowClear />;
+export default () => {
+  const {
+    token: { colorPrimary },
+  } = theme.useToken();
```
Comment (MadCcc) [inline components/color-picker/demo/allowClear.tsx:7]: demo 不改了，这样太复杂了

Comment (MadCcc) [reply components/color-picker/demo/allowClear.tsx:7]: 要么就直接把 defaultValue 设为 #1677ff

Comment (RedJue) [reply components/color-picker/demo/allowClear.tsx:7]: 行 那就写死色值好了

File: components/color-picker/demo/presets.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { ColorPicker } from 'antd';
 
 export default () => (
   <ColorPicker
+    defaultValue="#1677ff"
```
Comment (afc163) [inline components/color-picker/demo/presets.tsx:6]: 好像也没必要加 defaultValue，直接变成内置默认值就行。

Comment (yoyo837) [reply components/color-picker/demo/presets.tsx:6]: 应该指的是可以在没有value,(受控和非受控) 内部直接使用的默认值

Comment (RedJue) [reply components/color-picker/demo/presets.tsx:6]: 意思是默认选中预选色的第一个？

File: components/color-picker/demo/disabled.tsx (hunk 1 above)
Comment (crazyair) [reply components/color-picker/demo/disabled.tsx]: https://ant-design.antgroup.com/components/badge-cn#badge-demo-no-wrapper color 跟 token 是没有关系的，defaultValue 不一定要设主题色，可以设置随便颜色

File: components/color-picker/demo/presets.tsx (hunk 3 above)
Comment (crazyair) [reply components/color-picker/demo/presets.tsx:6]: 没有颜色应该是 无？要不加个灰底色，当做  empty？

Comment (afc163) [reply components/color-picker/demo/presets.tsx:6]: 确实应该是 `无` 更好。

Comment (zombieJ) [reply components/color-picker/demo/presets.tsx:6]: 不应该内置，输入型组件的默认值应该都是所见即所得的。否则对于表单提交场景，会和预期有出入：https://ant-design.antgroup.com/docs/blog/to-be-what-you-see-cn

**Comment by yoyo837** [comment]
应该留一个没有 `defaultValue` 的demo吧？, 或者 base里放俩？

```
<ColorPicker />;
<ColorPicker defaultValue="#1677ff" />;
```

**Comment by yoyo837** [comment]
Rebase 一下抛弃就可以抛弃这个 argos 。

**Comment by RedJue** [comment]
ok

**Comment by afc163** [comment]
这段时间没有 argos，心里有点慌。

**Comment by crazyair** [comment]
能看下无颜色，默认展示的效果吗

File: components/color-picker/demo/pure-panel.tsx [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
-import { ColorPicker, theme } from 'antd';
+import { ColorPicker } from 'antd';
 import type { Color } from 'antd/es/color-picker';
 
 const PureRenderColorPicker = ColorPicker._InternalPanelDoNotUseOrYouWillBeFired;
 
 export default () => {
-  const { token } = theme.useToken();
-  const [color, setColor] = useState<Color | string>(token.colorPrimary);
+  const [color, setColor] = useState<Color | string>('#1677ff');
```
Comment (kiner-tang) [inline components/color-picker/demo/pure-panel.tsx:8]: 我还是觉得这个类型我们可以包一层，如：`type AntColor = Color | string` 这样用户在使用的时候，就不需要那么麻烦了

**Comment by RedJue** [comment]
> 能看下无颜色，默认展示的效果吗

和已清除状态是一致的

**Review by yoyo837 — APPROVED** [review]

File: components/color-picker/demo/pure-panel.tsx (hunk 4 above)
Comment (RedJue) [reply components/color-picker/demo/pure-panel.tsx:8]: image
其实已经有了，只是 demo 忘改了。。

**Comment by crazyair** [comment]
> > 能看下无颜色，默认展示的效果吗
> 
> 和已清除状态是一致的

怎么清除？没看到 x https://ant-design.antgroup.com/components/color-picker-cn

File: components/progress/progress.tsx [hunk 5]
```diff
@@ -40,7 +40,7 @@ export interface ProgressProps extends ProgressAriaProps {
   type?: ProgressType;
   percent?: number;
   format?: (percent?: number, successPercent?: number) => React.ReactNode;
-  status?: (typeof ProgressStatuses)[number];
+  status?: typeof ProgressStatuses[number];
```
Comment (afc163) [inline components/progress/progress.tsx:43]: 不需要改其他组件。

Comment (RedJue) [reply components/progress/progress.tsx:43]: 好

**Pull request merged**